src/validationinterface.cpp

```diff
@@ -88,9 +88,11 @@ void RegisterValidationInterface(CValidationInterface* pwalletIn) {
 }
 
 void UnregisterValidationInterface(CValidationInterface* pwalletIn) {
-    if (g_signals.m_internals) {
-        g_signals.m_internals->m_connMainSignals.erase(pwalletIn);
-    }
+    AssertLockNotHeld(cs_main);
+    std::promise<void> promise;
```

Comment:
You don't need make promise or calls function if `g_signals.m_internals` is nullptr.

Reply:
Done.